### DISCUSSION TITLE
Possible to hide and restore side bar without tree reloading?

### DISCUSSION THREAD

**Discussion by fragande** [piroor/treestyletab#3154] [Q&A]
It seems that when hiding and restoring the side bar (by pressing F1) the whole tree reloads every time. As I have _a lot_ of unloaded tabs this takes a good 20-30s while RAM and CPU usage shoots through the roof. 

I don't really understand why the tree has to reload on hide/restore but can't find a setting for this. Is there one or some other way of solving this issue?

**Comment by irvinm** [comment]
Do you have:  Options -> Advanced -> "Optimize tree restoration via cache" enabled?  How many tabs are you talking about?  I don't even use the cache with around 500 tabs and it doesn't take more than a 4-5 seconds.  Maybe with "thousands" of tabs, I could see 20-30 seconds.

**Reply by fragande** [reply]
I have that enabled yes, but disabling it didn't make much of a difference. Currently I have about 800 tabs. Biggest slowdown is probably due to RAM filling up instantly (I only have 16 GB). I would like to be able to just visually hide the sidebar temporarily while watching a YouTube video for example, so even 4-5 seconds of the system grinding to a halt would be unbearable.

I know there's CSS to auto hide the sidebar. I would like to accomplish that but with a button press without reloading the tree.

**Reply by irvinm** [reply]
Ok, I totally understand what you are saying.  I see that TST is using browser.sidebarAction.toggle() [For FF 73+] and that seems to be the right way to do it, but I'm guessing there is some verification\initialization to ensure consistency with the original tab bar when showing the sidebar again.  @piroor will need to answer that.

The CSS solutions would certainly work, but there would be maintenance and upkeep when things change and that can be hard.

**Comment by piroor** [comment]
This is due to restrictions of Firefox itself. TST is now just a sidebar panel and Firefox always unloads sidebar panel contents provided by an addon when it is hidden.

There are some helper addons (ex. "TST Indent Line" and "TST Active Tab in Collapsed Tree" I made) which make initialization process of the sidebar contents slower. Recently I've introduce some changes to optimize their performance (not released yet), so please wait for their next releases.

**Reply by fragande** [reply]
I see, thank you for the insight. Wish I knew CSS so I could go that route, but the only readily available scripts out there are for pointer hover.
